cdcpd/tests/integration/static_rope.cpp

```diff
@@ -0,0 +1,142 @@
+{
+    if (PRINT_DEBUG_MESSAGES)
+    {
+        std::stringstream ss;
+        ss << initial_template_edges;
+        ROS_WARN("Initial template edges");
+        ROS_WARN(ss.str().c_str());
+    }
+
+    // Specify constant scalar and boolean parameters
+    float const objective_value_threshold = 1.0;
+    bool const use_recovery = false;
+    double const alpha = 0.5;
+    double const beta = 1.0;
+    double const lambda = 1.0;
+    double const k = 100.0;
+    float const zeta = 10.0;
+    float const obstacle_cost_weight = 0.001;
+    float const fixed_points_weight = 10.0;
+
+    CDCPD* cdcpd_ptr = new CDCPD(initial_tracked_points, initial_template_edges,
+        objective_value_threshold, use_recovery, alpha, beta, lambda, k, zeta, obstacle_cost_weight,
+        fixed_points_weight);
+
+    return cdcpd_ptr;
+}
+
+// Resimulate CDCPD on a set of previously recorded input point clouds in a bag file.
+PointCloud::Ptr resimulateCdcpd(CDCPD& cdcpd_sim,
+    std::vector<boost::shared_ptr<PointCloudRGB>> const& input_clouds,
+    PointCloud::Ptr const& initial_tracked_points, float const max_rope_length,
+    int const num_points)
+{
+    // Do some setup of parameters and gripper configuration.
+    PointCloud::Ptr tracked_points = initial_tracked_points;
+    ObstacleConstraints obstacle_constraints;  // No need to specify anything with this demo.
+    float const max_segment_length = max_rope_length / static_cast<float>(num_points);
+    unsigned int gripper_count = 0U;
+    smmap::AllGrippersSinglePose q_config;
+    const smmap::AllGrippersSinglePoseDelta q_dot{gripper_count, kinematics::Vector6d::Zero()};
+    Eigen::MatrixXi gripper_indices(1, gripper_count);
+
+    // Read the CDCPD input from the rosbag and step through the execution until the end of input is
+    // reached.
+    for (auto & cloud : input_clouds)
+    {
+        // Run a single "iteration" of CDCPD mimicing the points_callback lambda function found in
+        // cdcpd_node.cpp
+        CDCPD::Output out = cdcpd_sim(cloud, tracked_points, obstacle_constraints,
+            max_segment_length, q_dot, q_config, gripper_indices);
+        tracked_points = out.gurobi_output;
+
+        // Do a health check of CDCPD
+        if (out.status == OutputStatus::NoPointInFilteredCloud ||
+            out.status == OutputStatus::ObjectiveTooHigh)
+        {
+            std::string err_msg = "CDCPD failed for this iteration. We need to implement "
+                "cdcpd_node.cpp reset_if_bad lambda function!";
+            ROS_ERROR(err_msg.c_str());
+        }
+    }
+    return tracked_points;
+}
+
+TEST(StaticRope, testResimPointEquivalency)
+{
+    // Read in the ros bagfile that we'll be resimulating and checking CDCPD performance against.
+    rosbag::Bag bag;
+    // This test is executed from ~/.ros, so we have to do some funky path manipulation to get to
+    // the bag files
+    bag.open("../../../../../src/cdcpd/demos/rosbags/demo_1_static_rope.bag",
```

Comment:
consider using the rospack `https://wiki.ros.org/Packages#C.2B-.2B-` library to get the path to the `cdcpd` package

Reply:
Good catches on these. I've addressed them and will close pull request with a squash commit.